backend/controllers/user.controller.js

```diff
@@ -1,37 +1,70 @@
 const user = require('../models/user.model.js');
+const jwt = require('jsonwebtoken');
+require('dotenv').config();
+const bcrypt = require('bcrypt');
 
 const createUser = async (req,res)=>{
     try {
         const {username, password, email} = req.body;
     if (!username || !email || !password) {
       return  res.status(401).json({message: 'All fields are required'});
     };
-        const User = await user.create(req.body);
-        res.status(200).json(User)
+    const salt = await bcrypt.genSalt(10);
+    const hashedPassword = await bcrypt.hash(password, salt);
+
+
+        const User = await user.create({
```

Comment:
Let's make sure the username and email are not already in the db and also set them as unique in MongoDB to avoid duplicates.